lib/insights_cloud/async/connector_playbook_execution_reporter_task.rb

```diff
@@ -43,7 +43,7 @@ def rescue_strategy_for_self
       end
 
       def done?(current_status = invocation_status)
-        job_invocation.finished? || current_status.map { |_host_id, task_status| task_status['report_done'] }.all?
+        current_status[:task_state][:task_done_reported]
```

Comment:
Since the task output is serialized, we should make sure it's boolean:

```suggestion
        ActiveModel::Type::Boolean.new.cast(current_status[:task_state][:task_done_reported])
```